### PR TITLE
feature: bind_multiple_http_paths 

### PR DESCRIPTION
**Pull request by Gohlub**
## Problem

Refers to issue #101. This PR introduces another function to the HttpServer struct, which enables users to bind a vector of strings (representing http paths) to the server.

## Solution
I tried to make the function ergonomic (with the optional secure flag) though I am not sure whether any further configuration is needed (to handle filtering to which end-points should be secure, or not.

## Docs Update
I will update the docs once the merge is approved.

## Notes
This is my first pass at contributing to Kinode. I wanted to start simple and hope to do more contributions in the future. Let me know what you think. I didn't see any reference to how testing should be done so I did my own just to make sure things work as expected, and I am open to suggestions.

### CODE REVIEW

File: src/http/server.rs [hunk 1]
```diff
@@ -1048,6 +1048,48 @@ impl HttpServer {
...
+    /// Uses the security setting from HttpBindingConfig if secure is None,
+    /// forces secure binding if Some(true), or forces non-secure binding if Some(false).
+    /// All paths must be bound successfully, or none will be bound. If any path
+    /// fails to bind, all previously bound paths will be unbound before returning
+    /// the error.
+    pub fn bind_multiple_http_paths<T: Into<String>>(
+        &mut self,
+        paths: Vec<T>,
+        config: HttpBindingConfig,
+        secure: Option<bool>,
```
Comment (dr-frmr) [inline src/http/server.rs:1062]: I'm confused why the `secure` flag is necessary. `HttpBindingConfig` allows you to set whether the path will be a secure subdomain, so what's the point of an additional flag?

**Review by dr-frmr — COMMENTED** [review]
Looks good other than secure flag, can that be removed?

File: src/http/server.rs (hunk 1 above)
Comment (Gohlub) [reply src/http/server.rs:1062]: I suppose I over-engineered it. It slipped my mind that `HttpBindingConfig` already allows you to declare whether a path will be secure or not. I think I remember a time where that config object didn't exist (or maybe I was unaware of it) so it didn't occur naturally. In any case, I'll get rid of it and update the PR asap.

**Comment by Gohlub** [comment]
I removed the flag. Let me know if there is anything else.

**Review by dr-frmr — APPROVED** [review]
LGTM!

**Pull request merged**